include/boost/json/impl/object.hpp

```diff
@@ -518,19 +518,43 @@ namespace detail {
 unchecked_object::
 ~unchecked_object()
 {
-    if(! data_)
-        return;
     if(sp_.is_not_shared_and_deallocate_is_trivial())
         return;
-    value* p = data_;
-    while(size_--)
+
+    for( value* p = data_; p != end_; p += 2 )
```

Comment:
Why did you change this loop? I prefer the `while` form, as it puts each of the statements on their own line. The `for` is wider, and puts multiple statements on one line, which makes debugging and code coverage worse.

Reply:
I see why, because the loop is simpler.